### PR TITLE
fix: dfget couldn't download s3 directory correctly

### PR DESCRIPTION
**Pull request by piamo**
## Description

Something exception raised when using `dfget` to download a s3 directory.
Below is my usage:
```
dfget --recursive  --url "s3://my-bucket/my-dir" \
      --header "awsEndpoint: xxx" \
      --header ... \
      --output local_dir
```

I find the bug is that:

When download a s3 directory, client will call a s3 API `ListObjectsV2`, to check whether the url is a directory. And the API need a parameter `prefix` (used as prefix filter for s3 object ids, should not start with leading slash "/"). In dragonfly's s3 client code, the url will be parsed as `request.URL` and `request.URL.path` will be used as `prefix`(after add a trailing slash). Such `path` always have leading slash "/", e.g. '/', '/my-dir/', make `ListObjectsV2` API response wrong result.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by gaius-qi** [comment]
@piamo Please try to submit issues in English and fix lint.

**Comment by piamo** [comment]
> @piamo Please try to submit the pull request in English and fix lint.

lint这个错误是啥原因啊，感觉没改几行code啊-_-||
image

File: pkg/source/clients/s3protocol/s3_source_client.go [hunk 1]
```diff
@@ -286,3 +286,10 @@ func addTrailingSlash(s string) string {
 	}
 	return s + "/"
 }
+
+func buildListPrefix(s string) string {
```
Comment (yxxhero) [inline pkg/source/clients/s3protocol/s3_source_client.go:290]: add some unit tests for this func.

Comment (piamo) [reply pkg/source/clients/s3protocol/s3_source_client.go:290]: 额，`pkg/source/clients/s3protocol/`这个目录下还没有test文件，代码文件里的其他内部函数都没加ut，是不是需要整体搞一下-_-||

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**